api/cpp/include/slint-platform.h

```diff
@@ -695,6 +697,13 @@ class SoftwareRenderer : public AbstractRenderer
         // FIXME: Texture is currently opaque, but should be exposed
         virtual bool draw_texture(const DrawTextureArgs *texture, const PhysicalRegion &clip) = 0;
 
+        /// Fill the background of the buffer with the given brush.
+        virtual bool fill_background(const Brush &brush, const PhysicalRegion &clip) = 0;
+
+        /// Draw a rectangle specified by the DrawRectangleArgs. That rectangle must be clipped to
+        /// the given region
+        virtual bool draw_rectangle(const DrawRectangleArgs *args, const PhysicalRegion &clip) = 0;
```

Comment:
I must've overlooked this earlier, but: Is there a reason why we're using pointers here instead of a reference? We're not supporting a null pointer here, AFAICS.

Reply:
True. fixed.